testnet/stacks-node/src/neon_node.rs

```diff
@@ -958,6 +942,57 @@ impl InitializedNeonNode {
             )
         };
 
+        // has the tip changed from our previously-mined block for this epoch?
+        let attempt = {
```

Comment:
is the tip guaranteed to be ***better*** if it has changed?

Reply:
If you want your miner to build off of the canonical chain tip, then yes -- the last-mined block is getting compared to the current canonical Stacks chain tip, and if it was _not_ built on the current canonical Stacks chain tip, then the node re-builds the block.